gyft.py

```diff
@@ -126,31 +126,43 @@ def merge_slots(in_dict):
     for time in timetable_dict[day]:
         timetable_dict[day][time].append(" ")
         course_code = timetable_dict[day][time][0]
-        course_dept = course_code[:2]
+        course_dept = course_code[:2] 
         if course_dept not in courses.keys():
             courses[course_dept] = {}
         if course_code not in courses[course_dept].keys():
             courses[course_dept][course_code] = ''
 
 # scraping course names deptwise
 for dept in courses.keys():
-    DEPT_URL = COURSES_URL.format(dept)
-    r = s.get(DEPT_URL, headers=headers)
-    soup = bs(r.text, 'html.parser')
-    parentTable = soup.find('table', {'id': 'disptab'})
+    try: 
+        if dept == 'EP': # special case -> EP courses are under RJ department, not EP department
+            DEPT_URL = COURSES_URL.format('RJ')
+        else:
+            DEPT_URL = COURSES_URL.format(dept)
+        r = s.get(DEPT_URL, headers=headers)
+        soup = bs(r.text, 'html.parser')
+        parentTable = soup.find('table', {'id': 'disptab'})
+
+        rows = parentTable.find_all('tr')
 
-    rows = parentTable.find_all('tr')
+        for row in rows[1:]:
+            if 'bgcolor' in row.attrs:
+                continue 
+            cells = row.find_all('td')  
+            course_code = cells[0].text.strip()
+            course_name = cells[1].text.strip()
 
-    for row in rows[1:]:
-        if 'bgcolor' in row.attrs:
-            continue 
-        cells = row.find_all('td')  
-        course_code = cells[0].text.strip()
-        course_name = cells[1].text.strip()
 
-        if course_code in courses[dept]:
-            courses[dept][course_code] = course_name
-            logging.info(" {} - {}".format(course_code, course_name))
+            if course_code in courses[dept]:
+                courses[dept][course_code] = course_name
+                logging.info(" {} - {}".format(course_code, course_name))
+    except Exception as e:
+        print()
+        logging.error(" Error while scraping course names for course letter code {}".format(dept))
+        logging.error(e)
+        logging.error(" If you know the department code corresponding to the above code, please add it in the code after line 139, and submit a PR.")
```

Comment:
Two suggestions:
- Separate dict, so that you can refer to that dict's name
- Create a separate file containing that dict only

I would recommend the 2nd option

Reply:
I guess I will proceed with the 2nd option then, just like Rajiv also suggested